lib/matplotlib/patheffects.py

```diff
@@ -359,3 +362,153 @@ def draw_path(self, renderer, gc, tpath, affine, rgbFace):
         if clip_path:
             self.patch.set_clip_path(*clip_path)
         self.patch.draw(renderer)
+
+
+class TickedStroke(AbstractPathEffect):
+    """A line based PathEffect which draws a path with a ticked style."""
+    def __init__(self, offset=(0, 0),
+                 spacing=None, angle=None, length=None, flip=None, **kwargs):
+        """
+        Parameters
+        ----------
+        offset : pair of floats
+            The offset to apply to the path, in points.
+        spacing : float
+            The spacing between ticks in points.
+            Default is 10.0.
+        angle : float
+            The angle between the path and the tick in degrees.
+            Default is 45.0.
+        length : float
+            The length of the tick relative to spacing.
+            Default is 1.414 (sqrt(2)).  Recommend length=1.0 when angle=90.
+        flip : bool
+            Flag to flip the ticks to the other side of the line.
+            Default is False.
+        **kwargs
+            Extra keywords are stored and passed through to
+            :meth:`AbstractPathEffect._update_gc`.
+        """
+        super().__init__(offset)
+
+        if spacing is None:
+            spacing = 10.0
```

Comment:
This seems like a lot of code just to support passing None; I would not bother and just put the defaults in the signature.